src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java

```diff
@@ -95,23 +90,11 @@ protected ADBatchTaskCache(ADTask adTask) {
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
-            // same with dimension for opportunistic memory saving
-            // Usually, we use it as shingleSize(dimension). When a new point comes in, we will
-            // look at the point store if there is any overlapping. Say the previously-stored
-            // vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize
-            // overlapping x3, x4, and only store x5, x6.
-            .shingleSize(shingleSize)
+            // for external shingling, rcf does not recognize shingle
+            .shingleSize(1)
```

Comment:
Does this mean even user tune shingle size on frontend, we always use shingle size 1 ?

Reply:
shingle tuning will impact dimension.  It won't impact shingle size here, but impact the overall dimension of the data.

Reply:
added comments:

            // for external shingling, rcf does not recognize shingle.  Thus, shingle size
            // is 1 here.
            // shingle in detector config and shingle size here are different things.
            // shingle size in detector config impacts dimensions.

Reply:
Yes -- this is for existing models. This can be changed subsequently (in a single threaded mode) when a full shingle is available. However for a new model we should use the actual shingle size. In line 79
AnomalyDetector detector = adTask.getDetector(); are we getting config or the model? Seems like config, right?

Reply:
Ok, resolved. As lon g as we are using external shingling we should use shingleSize = 1 for the ThresholdedRCF model. If we ever switch to not have external shingle processing (which would be quite a bit of change) then we should revisit this.

Reply:
if pulling in PR278 for RCF, Revert back to shingleSIze(shingleSize). Even though shingling is external -- this will keep the thresholded model closer to its final state and remove this distinction.